crates/corro-admin/src/lib.rs

```diff
@@ -520,6 +523,61 @@ async fn handle_conn(
 
                     send_success(&mut stream).await;
                 }
+                Command::CompactEmpties => {
+                    let actor_ids: Vec<_> = {
+                        let r = bookie.read("admin compact empties").await;
+                        r.keys().copied().collect()
+                    };
+
+                    let (tx, mut rx) = mpsc::channel(4);
+                    let (done_tx, done_rx) = oneshot::channel::<Option<String>>();
+
+                    let agent = agent.clone();
+                    let started = std::time::Instant::now();
+                    tokio::task::spawn(async move {
+                        for actor_id in actor_ids {
+                            if let Err(e) =
+                                util::clear_empty_versions(agent.clone(), actor_id, Some(&tx)).await
+                            {
+                                done_tx.send(Some(format!("{e}"))).unwrap();
+                                return;
+                            }
+                        }
+
+                        done_tx.send(None).unwrap();
+                    });
+
+                    while let Some(msg) = rx.recv().await {
+                        info_log(&mut stream, msg).await;
+                    }
+
+                    match done_rx.await {
+                        Ok(None) => {
+                            let elapsed = started.elapsed().as_secs_f64();
+                            info_log(&mut stream, format!(
+                                    "Finished compacting empty versions!  Took {} seconds ({} minutes)",
+                                    elapsed,
+                                    elapsed / 60.0
+                                ),
+                            ).await;
+                            send_success(&mut stream).await
+                        }
+                        Ok(Some(err)) => {
+                            send_error(
+                                &mut stream,
+                                format!("An error occured while compacting empties: {err}"),
+                            )
+                            .await
+                        }
+                        _ => {
+                            send_error(
+                                &mut stream,
+                                "Failed to compact empties (check node logs for details)",
+                            )
+                            .await
+                        }
+                    }
```

Comment:
It doesn't seem like there's any reason to use channels for communicating with the spawned task.

```suggestion
                    let agent = agent.clone();
                    let started = std::time::Instant::now();
                    let done = tokio::task::spawn(async move {
                        for actor_id in actor_ids {
                            if let Err(e) =
                                util::clear_empty_versions(agent.clone(), actor_id, Some(&tx)).await
                            {
                                return Some(format!("{e}"));
                            }
                        }

                        None
                    });

                    while let Some(msg) = rx.recv().await {
                        info_log(&mut stream, msg).await;
                    }

                    match done.await {
                        Ok(None) => {
                            let elapsed = started.elapsed().as_secs_f64();
                            info_log(&mut stream, format!(
                                    "Finished compacting empty versions!  Took {} seconds ({} minutes)",
                                    elapsed,
                                    elapsed / 60.0
                                ),
                            ).await;
                            send_success(&mut stream).await
                        }
                        Ok(Some(err)) => {
                            send_error(
                                &mut stream,
                                format!("An error occured while compacting empties: {err}"),
                            )
                            .await
                        }
                        _ => {
                            send_error(
                                &mut stream,
                                "Failed to compact empties (check node logs for details)",
                            )
                            .await
                        }
                    }
```